inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -263,12 +264,15 @@ def refextract(obj, eng):
     pdf_references, text_references = [], []
     source = get_value(obj.data, 'acquisition_source.source')
 
-    pdf = get_pdf_in_workflow(obj)
-    if pdf:
+    tmp_pdf = get_pdf_in_workflow(obj)
+    if tmp_pdf:
         try:
-            pdf_references = extract_references_from_pdf(pdf, source)
+            pdf_references = extract_references_from_pdf(tmp_pdf, source)
         except TimeoutError:
             obj.log.error('Timeout when extracting references from PDF.')
+        finally:
+            if os.path.exists(tmp_pdf):
+                os.unlink(tmp_pdf)
```

Comment:
Is this safe? Aren't we deleting files from under invenio-files?

Reply:
the`tmp_pdf` is the locally downloaded copy of the pdf, not really the original one.